CMakeLists.txt

```diff
@@ -185,7 +185,7 @@ else()
   if(WIN32)
     add_compile_flag("-D_GNU_SOURCE")
     add_link_flag("-Wl,--stack,8388608")
-  else()
+  elseif(NOT EMSCRIPTEN)
     add_compile_flag("-fPIC")
```

Comment:
What is this flag even doing here in the first place?    CMake should be adding this automatically when it knows its building a DLL, no?   I'm OK with landing this part of the change for now, but would like to look into removing this line completely

Reply:
Might this have something to do with [this warning](https://travis-ci.org/WebAssembly/binaryen/jobs/639415666#L253)?

```
CMake Warning (dev) at CMakeLists.txt:272 (add_library):
  ADD_LIBRARY called with SHARED option but the target platform does not
  support dynamic linking.  Building a STATIC library instead.  This may lead
  to problems.
```

Reply:
I think we not be trying to build with SHARED using emscripten, as the warning says.

Other platforms can use SHARED but the emscripten build should only use STATIC.